### PR TITLE
HADOOP-17234. Add .asf.yaml to allow Github to Jira integration.

### PR DESCRIPTION
**Pull request by ayushtkn**
https://issues.apache.org/jira/browse/HADOOP-17234

### CODE REVIEW

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   3m 35s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  yamllint  |   0m  0s |  yamllint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ trunk Compile Tests _ |
| +1 :green_heart: |  shadedclient  |  21m  3s |  branch has no errors when building and testing our client artifacts.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedclient  |  16m  8s |  patch has no errors when building and testing our client artifacts.  |
||| _ Other Tests _ |
| -1 :x: |  asflicense  |   0m 29s |  The patch generated 1 ASF License warnings.  |
|  |   |  43m  1s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2253/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/2253 |
| Optional Tests | dupname asflicense yamllint |
| uname | Linux a2a9a15cb42a 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / d1c60a53f60 |
| asflicense | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2253/1/artifact/out/patch-asflicense-problems.txt |
| Max. process+thread count | 307 (vs. ulimit of 5500) |
| modules | C: . U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2253/1/console |
| versions | git=2.17.1 maven=3.6.0 |
| Powered by | Apache Yetus 0.13.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   3m 58s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  yamllint  |   0m  0s |  yamllint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ trunk Compile Tests _ |
| +1 :green_heart: |  shadedclient  |  23m 52s |  branch has no errors when building and testing our client artifacts.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedclient  |  17m 19s |  patch has no errors when building and testing our client artifacts.  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 37s |  The patch does not generate ASF License warnings.  |
|  |   |  47m 43s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2253/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/2253 |
| Optional Tests | dupname asflicense yamllint |
| uname | Linux d3b09e41508b 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / d1c60a53f60 |
| Max. process+thread count | 311 (vs. ulimit of 5500) |
| modules | C: . U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2253/2/console |
| versions | git=2.17.1 maven=3.6.0 |
| Powered by | Apache Yetus 0.13.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

File: .asf.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+github:
+  enabled_merge_buttons:
+    squash: true
+    merge: false
+    rebase: false
+notifications:
+  commits:      common-commits@hadoop.apache.org
+  issues:       common-issues@hadoop.apache.org
+  pullrequests: common-issues@hadoop.apache.org
+  jira_options: link label worklog
```
Comment (liuml07) [inline .asf.yaml:25]: nit: do we need newline at the end of file?

Comment (ayushtkn) [reply .asf.yaml:25]: The link was added by me only. But I got this checked from the Infra team on the asfInfra slack channel and they only told, post this things should work.

Regarding the newline, I don't know if it is required, Most of the files don't have it. If you say I will push a commit adding one

Comment (liuml07) [reply .asf.yaml:25]: No it's not required. It's not a blank line newline, but a hidden char. Sharing some thoughts:

1. No end of line char can lead to unexpected behavior when using other tools on the file. For e.g. git diff or external diff tools depend on newlines to generate a correct diff. `wc` command counting lines may also be broken. More can be found at [this discussion](https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline/729795#729795).
2. Next time when someone edits the file, some editors will automatically add a "newline" char at the end of this file - for good intention. If so, seems like they "changed" the last line of this file in their PR which they do not.

But that is not a real problem here. We can commit without fixing that, I'm fine with that.

**Comment by liuml07** [comment]
Maybe unrelated to this, but I see everytime we comment on a PR, the robot will update the `Time Tracking` field of the JIRA. Sometimes I see email notifications about tis. However this seems not useful yet to the community. Is it possible to disable it? Thanks!

Screen Shot 2020-08-27 at 1 45 02 PM

**Comment by ayushtkn** [comment]
@liuml07 Time tracking seems to enabled by jira, To disable that, I think we need Jira Admin rights.

https://confluence.atlassian.com/adminjiraserver073/configuring-time-tracking-861253946.html?_ga=2.77528598.2074936245.1598575483-27446335.1598575483

Which I think I don’t have, I can check with infra on this. One probable fix  “can be” to remove ‘worklog ‘ from ‘jira_options’ , but that will even disable comments on PR getting notified on the Jira and project summary.

**Pull request merged**